packages/k8s/src/k8s/index.ts

```diff
@@ -516,27 +517,22 @@ class BackOffManager {
 export function containerPorts(
   container: ContainerInfo
 ): k8s.V1ContainerPort[] {
-  // 8080:8080/tcp
-  const portFormat = /(\d{1,5})(:(\d{1,5}))?(\/(tcp|udp))?/
-
   const ports: k8s.V1ContainerPort[] = []
   for (const portDefinition of container.portMappings) {
-    const submatches = portFormat.exec(portDefinition)
+    const submatches = PORT_REGEXP.exec(portDefinition)
     if (!submatches) {
       throw new Error(
         `Port definition "${portDefinition}" is in incorrect format`
       )
     }
     const port = new k8s.V1ContainerPort()
-    port.hostPort = Number(submatches[1])
-    if (submatches[3]) {
+    if (Number(submatches[3])) {
```

Comment:
Should I first validate `submatches[1]` ? submatches[1] should always have a valid number...

Reply:
It might make sense to just throw a well known error message and fail the job. Otherwise the k8s api will fail us and the visibility on that error message is really poor.

We shouldn't encounter those scenarios, but it is just user input from the yaml, so its better to be safe then sorry

"Unexpected docker port format %s"